src/main/java/com/alexlasota/medicalclinic/controller/PatientController.java

```diff
@@ -0,0 +1,76 @@
+package com.alexlasota.medicalclinic.controller;
+
+import com.alexlasota.medicalclinic.model.PatientDto;
+import com.alexlasota.medicalclinic.service.PatientService;
+import com.alexlasota.medicalclinic.model.Patient;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("/patients")
+@RequiredArgsConstructor
+public class PatientController {
+
+    private final PatientService patientService;
+
+    @GetMapping
+    public List<PatientDto> getPatients() {
+        return patientService.getPatients()
+                .stream()
+                .map(this::convertToDto)
+                .collect(Collectors.toList());
+    }
+
+    //w sytuacji kiedy w sciezce nasz parametr nazywa sie tak samo jak zmienna
+    // to wtedy nie musimy explicit podawac jej nazwy @PathVariable("email")
+    @GetMapping("/{email}")
+    public PatientDto getPatientByEmail(@PathVariable String email) {
+        return convertToDto(patientService.getPatientByEmail(email));
+    }
+    //W rezultacie, gdy żądanie GET zostanie wysłane do endpointu /
+    // z adresem e-mail jako częścią ścieżki (np. /patients/john@example.com),
+    // metoda getPatientByEmail zostanie wywołana,
+    // a pacjent o podanym adresie e-mail zostanie zwrócony jako odpowiedź na żądanie.
+
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public void addPatient(@RequestBody Patient patient) {
```

Comment:
Tutaj moim zdaniem powinnismy zwracac Pacjenta powołam się tutaj na pole id, które jest dodawane przez aplikację -> w sytuacji jesli mielbysmy integrację aplikacji A z aplikacją B to aplikacja A wysyłać request o utworzenie nowego pacjenta nie dowiedziałaby się jaki id został nadany pacjentowi, którego chciała utworzyć.